### ISSUE TITLE
Allowing user to manually set slug in admin module

### ISSUE THREAD

**Issue by mungurs** [arbory/arbory#9]
There are scenarios where user is allowed to manually edit slug input in modules. Slug field is set as optional, so that user can leaves empty slug field and depends on auto-generated slug.  When user leaves empty field and tries to save model Spatie\Sluggable\HasSlug dies with error:
Type error: Return value of App\Products\ProductTranslation::generateNonUniqueSlug() must be of the type string, null returned.

Digging deeper I found that [hasCustomSlugBeenUsed ](https://github.com/spatie/laravel-sluggable/blob/master/src/HasSlug.php#L104) method does not check if the new slug is empty or set at all. 

Do we override this logic in model or  in arbory trait? Or there is a better solution?
The fix would be easy
```
    protected function hasCustomSlugBeenUsed(): bool
    {
        $slugField = $this->slugOptions->slugField;
        if (empty($this->$slugField)) {
               return false;
        }
        return $this->getOriginal($slugField) != $this->$slugField;
    }
```
Only drawback is that this would forces non-empty slugs.

**Comment by strads10** [comment]
The correct solution would be to use the [slug field](https://github.com/arbory/arbory/blob/master/src/Admin/Form/Fields/Slug.php), unfortunately as of now its closely coupled with nodes, and needs to be updated to work with other modules again. Accepting PR on this.

**Comment by strads10** [comment]
Slug field can now be used in other modules, resolved in ab484fa4d5e84ed65beac89421cbf56dba5c7093. The second argument specifies from which field the slug should be created from.
